release-19.2: backupccl: allow restoration of empty db

pbardea opened the pull request:
Backport 1/1 commits from #42005.

/cc @cockroachdb/release

---

Previously, RESTORE did not support restoring empty databases. However,
there are use cases in which scripts perform the backup and restore
process and they have run into issues since when they go to RESTORE all
of their databases, if any are empty the script will fail.

Since BACKUP supports the backing up of empty databases, it makes sense
to allow RESTORE to restore those databases that were backed up.

Fixes #41962.

Release note (enterprise change): RESTORE now supports the restoration
of empty databases.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42733)
<!-- Reviewable:end -->

pbardea commented:
bors r+

pbardea committed:
backupccl: allow restoration of empty db

Previously, RESTORE did not support restoring empty databases. However,
there are use cases in which scripts perform the backup and restore
process and they have run into issues since when they go to RESTORE all
of their databases, if any are empty the script will fail.

Since BACKUP supports the backing up of empty databases, it makes sense
to allow RESTORE to restore those databases that were backed up.

Release note (enterprise change): RESTORE now supports the restoration
of empty databases.
